### PR TITLE
Sort languages alphabetically depending on current language selected

### LINKED ISSUE #68: The "Settings -> Language" menu should order options alphabetically
See: https://github.com/Mosquito-Alert/Mosquito-Alert-Mobile-App/blob/6c5993a230a86f958c8dca8bcfef2994a6b93ebe/lib/pages/settings_pages/settings_page.dart#L301-L334

### PR DESCRIPTION
**Pull request by DigitalSeneca**
fixes #68 

| English | Spanish | Basque |
| ------------- | ------------- | ------------- |
| image  | image | image |

### CODE REVIEW

File: lib/pages/settings_pages/settings_page.dart [hunk 1]
```diff
@@ -298,12 +296,25 @@ class _SettingsPageState extends State<SettingsPage> {
     );
   }
 
+  languagesSortedAlphabetically() {
```
Comment (epou) [inline lib/pages/settings_pages/settings_page.dart:299]: I'd add the return type in front of the function and two arguments, not to depend on variables defined in the file itself
`List<Language> languagesSortedAlphabetically(BuildContext context, List<Language> languageCodes)`

File: lib/pages/settings_pages/settings_page.dart [hunk 2]
```diff
@@ -298,12 +296,25 @@ class _SettingsPageState extends State<SettingsPage> {
...
+    // Alphabet depends on the currently selected language
+    List<Language> sortedLanguages = languageCodes.map((language) {
+      String translatedName = MyLocalizations.of(context, language.isoCode) ?? language.name;;
```
Comment (epou) [inline lib/pages/settings_pages/settings_page.dart:302]: Change double `;;` to single `;` at the end of the line

File: lib/pages/settings_pages/settings_page.dart [hunk 3]
```diff
@@ -298,12 +296,25 @@ class _SettingsPageState extends State<SettingsPage> {
...
+    return sortedLanguages;
+  }
+
   void _openLanguagePickerDialog() => showDialog(
         context: context,
         builder: (context) => Theme(
             data: Theme.of(context).copyWith(primaryColor: Style.colorPrimary),
             child: LanguagePickerDialog(
-                languages: languageCodes..sort(((a, b) => a.name.compareTo(b.name))),
+                languages: languagesSortedAlphabetically(),
```
Comment (epou) [inline lib/pages/settings_pages/settings_page.dart:317]: If adding arguments call as:
`languagesSortedAlphabetically(context, languageCodes);`

File: lib/pages/settings_pages/settings_page.dart [hunk 4]
```diff
@@ -298,12 +296,25 @@ class _SettingsPageState extends State<SettingsPage> {
...
+    return sortedLanguages;
+  }
+
   void _openLanguagePickerDialog() => showDialog(
         context: context,
         builder: (context) => Theme(
             data: Theme.of(context).copyWith(primaryColor: Style.colorPrimary),
             child: LanguagePickerDialog(
-                languages: languageCodes,
+                languages: languagesSortedAlphabetically(),
```
Comment (epou) [inline lib/pages/settings_pages/settings_page.dart:317]: If this function does not have to be used anywhere else, I would go for a one-line solution. I was thinking something like:

```dart
languages: languageCodes.map(
	(language) => Language(
		language.isoCode,
		MyLocalizations.of(context, language.isoCode) ?? language.name
	)).toList()..sort(
	        (a, b) => a.name.compareTo(b.name)
        );
```

File: lib/pages/settings_pages/settings_page.dart (hunk 2 above)
Comment (DigitalSeneca) [reply lib/pages/settings_pages/settings_page.dart:302]: sorry for the typo!

**Review by epou — APPROVED** [review]

**Pull request merged**